Source/Dafny/ConstructorWarning.cs

```diff
@@ -0,0 +1,53 @@
+using System.Linq;
+using System.Reactive;
+using Microsoft.Dafny;
+
+/*
+   * Code Example that displays the Warnings
+   *
+   * datatype Color = Red | Green | ShadesOfGray(nat)
```

Comment:
For your example below, besides the option `/warnMissingConstructorParenthesis` could you also always add the warning in the case when:
```suggestion
   * datatype Color = Red | Green() | ShadesOfGray(nat)
```
then there is ALWAYS a warning if there are missing parentheses in pattern `Green` ?

Reply:
This implies a bit more work, and I suggest we leave that for another PR for the sake of time.